opencog/data/table/table.cc

```diff
@@ -905,7 +906,7 @@ void complete_truth_table::populate(const Handle &handle)
 	// convert Links in the result of the interpreter to bool,
 	// and store it to the truth table.
 	std::transform(result.begin(), result.end(), begin(), [](ValuePtr p){
-		return HandleCast(p)->get_type() == TRUE_LINK;
+		return bool_value_to_bool(p);
```

Comment:
I think you should be able to pass `bool_value_to_bool` directly without having to wrap it in a lambda.